quolance-api/src/main/java/com/quolance/quolance_api/services/entity_services/impl/BlogPostServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+
+import com.quolance.quolance_api.dtos.blog.BlogPostRequestDto;
+import com.quolance.quolance_api.dtos.blog.BlogPostResponseDto;
+import com.quolance.quolance_api.entities.BlogPost;
+import com.quolance.quolance_api.entities.User;
+import com.quolance.quolance_api.repositories.BlogPostRepository;
+import com.quolance.quolance_api.repositories.UserRepository;
+import com.quolance.quolance_api.services.entity_services.BlogPostService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class BlogPostServiceImpl implements BlogPostService {
+
+    private final BlogPostRepository blogPostRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public BlogPostResponseDto create(@Valid BlogPostRequestDto request) {
+        // Fetch user
+        User user = userRepository.findById(request.getUserId())
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        // Create and save the blog post
+        BlogPost blogPost = BlogPost.builder()
+                .content(request.getContent())
+                .user(user)
+                .dateCreated(LocalDateTime.now())
+                .build();
+
+        BlogPost savedBlogPost = blogPostRepository.save(blogPost);
+
+        // Return response DTO
+        return mapToResponseDto(savedBlogPost);
+    }
+
+    @Override
+    public List<BlogPostResponseDto> getAll() {
+        return blogPostRepository.findAll().stream()
+                .map(this::mapToResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Optional<BlogPost> findById(Long id) {
+        return blogPostRepository.findById(id);
+    }
+
+    @Override
+    public List<BlogPostResponseDto> getBlogPostsByUserId(Long userId) {
+        // Fetch all blog posts for the given user ID
+        List<BlogPost> blogPosts = blogPostRepository.findByUserId(userId);
+
+        // Map blog posts to response DTOs
+        return blogPosts.stream()
+                .map(this::mapToResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BlogPostResponseDto update(Long id, @Valid BlogPostRequestDto request) {
+        // Fetch the existing blog post
+        BlogPost blogPost = blogPostRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Blog post not found"));
+
+        // Update the blog post fields
+        blogPost.setContent(request.getContent());
+
+        // Save the updated blog post
+        BlogPost updatedBlogPost = blogPostRepository.save(blogPost);
+
+        // Return response DTO
+        return mapToResponseDto(updatedBlogPost);
+    }
+
+    @Override
+    public void delete(Long id) {
```

Comment:
need to check that the post belongs to the person who deletes. Pass in the user id or the user and do the check.